alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php

```diff
@@ -473,13 +474,17 @@ private function getAutoEmbedCode($targetId = null)
 
 		$entry_id = $this->getRequestParameter(self::ENTRY_ID_PARAM_NAME);
 		$playlist_id = $this->getRequestParameter(self::PLAYLIST_ID_PARAM_NAME);
+		$external_source = $this->getRequestParameter(self::EXTERNAL_SOURCE_PARAM_NAME);
 		$loadContentMethod = "";
 		if (!is_null($entry_id)) {
 		    $loadContentMethod = "kalturaPlayer.loadMedia({\"entryId\":\"$entry_id\"});";
 		} elseif (!is_null($playlist_id)) {
 		    $loadContentMethod = "kalturaPlayer.loadPlaylist({\"playlistId\":\"$playlist_id\"});";
-		} else {
-		    KExternalErrors::dieError(KExternalErrors::MISSING_PARAMETER, "Entry and Playlist ID not defined");
+		} elseif (!is_null($external_source)) {
+			$loadContentMethod = "kalturaPlayer.setMedia({\"sources\":$external_source});";
+		}
+		else {
+			KExternalErrors::dieError(KExternalErrors::MISSING_PARAMETER, "Entry and Playlist ID not defined");
```

Comment:
maybe should be now more general message, something like playback source id not defined